src/api/users.ts

```diff
@@ -0,0 +1,38 @@
+import { Comment } from '../types/Comment';
+import { Post } from '../types/Post';
+import { User } from '../types/User';
+import { client } from '../utils/fetchClient';
+
+export const getUsers = () => {
+  return client.get<User[]>(`/users`);
+};
+
+export const getPostsByUser = (userId = 0) => {
+  return client.get<Post[]>(`/posts?userId=${userId}`);
+};
+
+export const getCommentsByPost = (postId = 0) => {
+  return client.get<Comment[]>(`/comments?postId=${postId}`);
+};
+
+export const addComment = ({
+  postId,
+  name,
+  email,
+  body,
+}: Omit<Comment, 'id'>) => {
+  return client.post<Comment>('/comments', {
+    postId,
+    name,
+    email,
+    body,
+  });
+};
+
+export const deleteTodos = (todoId: number) => {
+  return client.delete(`/todos/${todoId}`);
+};
```

Comment:
The function `deleteTodos` appears to be unrelated to the current context of users, posts, and comments. Consider removing it if it's not needed.